discovery/docker_discovery.go

```diff
@@ -53,10 +55,29 @@ func NewDockerDiscovery(endpoint string, svcNamer ServiceNamer, ip string) *Dock
 	return &discovery
 }
 
+func checkDockerSocket(endpoint string) {
+	socketPath := strings.TrimPrefix(endpoint, "unix://")
+	info, err := os.Stat(socketPath)
+	if os.IsNotExist(err) {
+		log.Errorf("Docker socket not found at %s", socketPath)
+	} else if err != nil {
+		log.Errorf("Error checking %s: %v", socketPath, err)
+	} else {
+		mode := info.Mode()
+		if mode&os.ModeSocket == 0 {
+			log.Warnf("%s exists but is not a socket", socketPath)
+		} else {
+			log.Debugf("Docker socket exists at %s with permissions %v", socketPath, mode.Perm())
+		}
+	}
```

Comment:
```suggestion
		log.Errorf("Error checking %s: %v", socketPath, err)
		return
	}
	mode := info.Mode()
	if mode&os.ModeSocket == 0 {
		log.Warnf("%s exists but is not a socket", socketPath)
		return
	}
	
	log.Debugf("Docker socket exists at %s with permissions %v", socketPath, mode.Perm())
```

Reply:
@relistan wouldn't we need to return on line 62 as well with this suggestion?